### ISSUE TITLE
Migration issue

### ISSUE THREAD

**Issue by sunscreem** [putyourlightson/craft-sherlock#7]
Small issue today when I applied the latest version 2.1.1

I have my staging server as `'allowAdminChanges'   => false,`

Toggled that and the migration ran ok.

```
$ ./craft  migrate/all
Yii Migration Tool (based on Yii v2.0.19)

Exception: Changes to the project config are not possible while in read-only mode. (/home/forge/**/vendor/craftcms/cms/src/services/ProjectConfig.php:386)
#0 /home/forge/**/vendor/craftcms/cms/src/services/Plugins.php(702): craft\services\ProjectConfig->set('plugins.sherloc...', Array)
#1 /home/forge/**/vendor/putyourlightson/craft-sherlock/src/migrations/m190614_120000_update_duration_settings.php(30): craft\services\Plugins->savePluginSettings(Object(putyourlightson\sherlock\Sherlock), Array)
#2 /home/forge/**/vendor/craftcms/cms/src/db/Migration.php(56): putyourlightson\sherlock\migrations\m190614_120000_update_duration_settings->safeUp()
#3 /home/forge/**/vendor/craftcms/cms/src/db/MigrationManager.php(243): craft\db\Migration->up(true)
#4 /home/forge/**/vendor/craftcms/cms/src/db/MigrationManager.php(163): craft\db\MigrationManager->migrateUp(Object(putyourlightson\sherlock\migrations\m190614_120000_update_duration_settings))
#5 /home/forge/**/vendor/craftcms/cms/src/services/Updates.php(225): craft\db\MigrationManager->up()
#6 /home/forge/**/vendor/craftcms/cms/src/console/controllers/MigrateController.php(252): craft\services\Updates->runMigrations(Array)
#7 [internal function]: craft\console\controllers\MigrateController->actionAll()
#8 /home/forge/**/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#9 /home/forge/**/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#10 /home/forge/**/vendor/yiisoft/yii2/console/Controller.php(148): yii\base\Controller->runAction('all', Array)
#11 /home/forge/**/vendor/yiisoft/yii2/base/Module.php(528): yii\console\Controller->runAction('all', Array)
#12 /home/forge/**/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction('migrate/all', Array)
#13 /home/forge/**/vendor/craftcms/cms/src/console/Application.php(93): yii\console\Application->runAction('migrate/all', Array)
#14 /home/forge/**/vendor/yiisoft/yii2/console/Application.php(147): craft\console\Application->runAction('migrate/all', Array)
#15 /home/forge/**/vendor/yiisoft/yii2/base/Application.php(386): yii\console\Application->handleRequest(Object(craft\console\Request))
#16 /home/forge/**/craft(32): yii\base\Application->run()
#17 {main}
Exception 'craft\errors\MigrateException' with message 'An error occurred while migrating Sherlock.'

in /home/forge/**/vendor/craftcms/cms/src/services/Updates.php:233

Stack trace:
#0 /home/forge/**/vendor/craftcms/cms/src/console/controllers/MigrateController.php(252): craft\services\Updates->runMigrations(Array)
#1 [internal function]: craft\console\controllers\MigrateController->actionAll()
#2 /home/forge/**/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#3 /home/forge/**/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#4 /home/forge/**/vendor/yiisoft/yii2/console/Controller.php(148): yii\base\Controller->runAction('all', Array)
#5 /home/forge/**/vendor/yiisoft/yii2/base/Module.php(528): yii\console\Controller->runAction('all', Array)
#6 /home/forge/**/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction('migrate/all', Array)
#7 /home/forge/**/vendor/craftcms/cms/src/console/Application.php(93): yii\console\Application->runAction('migrate/all', Array)
#8 /home/forge/**/vendor/yiisoft/yii2/console/Application.php(147): craft\console\Application->runAction('migrate/all', Array)
#9 /home/forge/**/vendor/yiisoft/yii2/base/Application.php(386): yii\console\Application->handleRequest(Object(craft\console\Request))
#10 /home/forge/**/craft(32): yii\base\Application->run()
#11 {main}
```

**Comment by putyourlightson** [comment]
I'm not quite following. Did the migration run ok or result in errors?

**Comment by sunscreem** [comment]
Sorry Ben, I should have explained better.

Ran the composer update to for todays release locally, and all good.

Then deployed to staging and get a deployment error.

Turns out it was the migrate/all that was failing, so I ran it manually on the server and received that output shown above.

I toggled `allowAdminChanges` to true, I then ran `./craft migrate/all` from the command line and all good.

So it looks like its attempting to alter the config during the migration?
